ovos_utils/skills/locations.py

```diff
@@ -65,11 +73,18 @@ def get_skill_directories(config=None) -> List[str]:
     conf = conf.get("skills") or {}
     # extra_directories is a list of directories containing skill subdirectories
     # NOT a list of individual skill folders
-    skill_locations += conf.get("extra_directories") or []
+    # preserve order while removing any duplicate entries
+    extra_dirs = [expanduser(d) for d in
+                  list(dict.fromkeys(conf.get("extra_directories")))
```

Comment:
whis is this `list(dict.fromkeys`? `conf.get("extra_directories", [])`should be much simpler?

theres also a lot of expandirs here, maybe it would be cleaner to extract this into a var and filter afterwards

Reply:
This was the most efficient way I found to remove duplicates and maintain element order, since there is some order dependency for skill loading.. We probably could do this in 2 steps instead of 1 to clean up the number of `expanduser` calls though..